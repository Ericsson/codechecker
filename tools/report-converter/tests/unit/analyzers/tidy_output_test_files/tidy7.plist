<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>diagnostics</key>
	<array>
		<dict>
			<key>category</key>
			<string>modernize</string>
			<key>check_name</key>
			<string>modernize-use-nullptr</string>
			<key>description</key>
			<string>use nullptr</string>
			<key>issue_hash_content_of_line_in_context</key>
			<string>5385d8fae5f3551df1dcb8e0456b204f</string>
			<key>location</key>
			<dict>
				<key>col</key>
				<integer>20</integer>
				<key>file</key>
				<integer>0</integer>
				<key>line</key>
				<integer>14</integer>
			</dict>
			<key>notes</key>
			<array>
				<dict>
					<key>location</key>
					<dict>
						<key>col</key>
						<integer>20</integer>
						<key>file</key>
						<integer>0</integer>
						<key>line</key>
						<integer>14</integer>
					</dict>
					<key>message</key>
					<string>nullptr (fixit)</string>
				</dict>
			</array>
			<key>path</key>
			<array>
				<dict>
					<key>depth</key>
					<integer>0</integer>
					<key>kind</key>
					<string>event</string>
					<key>location</key>
					<dict>
						<key>col</key>
						<integer>20</integer>
						<key>file</key>
						<integer>0</integer>
						<key>line</key>
						<integer>14</integer>
					</dict>
					<key>message</key>
					<string>use nullptr</string>
				</dict>
			</array>
			<key>type</key>
			<string>clang-tidy</string>
		</dict>
		<dict>
			<key>category</key>
			<string>clang</string>
			<key>check_name</key>
			<string>clang-diagnostic-array-bounds</string>
			<key>description</key>
			<string>array index 10 is past the end of the array (that has type 'int[5]')</string>
			<key>issue_hash_content_of_line_in_context</key>
			<string>f5dfd26909c583aeeeb8f0920b5c0366</string>
			<key>location</key>
			<dict>
				<key>col</key>
				<integer>5</integer>
				<key>file</key>
				<integer>0</integer>
				<key>line</key>
				<integer>19</integer>
			</dict>
			<key>path</key>
			<array>
				<dict>
					<key>edges</key>
					<array>
						<dict>
							<key>end</key>
							<array>
								<dict>
									<key>col</key>
									<integer>5</integer>
									<key>file</key>
									<integer>0</integer>
									<key>line</key>
									<integer>19</integer>
								</dict>
								<dict>
									<key>col</key>
									<integer>5</integer>
									<key>file</key>
									<integer>0</integer>
									<key>line</key>
									<integer>19</integer>
								</dict>
							</array>
							<key>start</key>
							<array>
								<dict>
									<key>col</key>
									<integer>5</integer>
									<key>file</key>
									<integer>0</integer>
									<key>line</key>
									<integer>17</integer>
								</dict>
								<dict>
									<key>col</key>
									<integer>5</integer>
									<key>file</key>
									<integer>0</integer>
									<key>line</key>
									<integer>17</integer>
								</dict>
							</array>
						</dict>
					</array>
					<key>kind</key>
					<string>control</string>
				</dict>
				<dict>
					<key>depth</key>
					<integer>0</integer>
					<key>kind</key>
					<string>event</string>
					<key>location</key>
					<dict>
						<key>col</key>
						<integer>5</integer>
						<key>file</key>
						<integer>0</integer>
						<key>line</key>
						<integer>17</integer>
					</dict>
					<key>message</key>
					<string>array 'arr' declared here</string>
				</dict>
				<dict>
					<key>depth</key>
					<integer>0</integer>
					<key>kind</key>
					<string>event</string>
					<key>location</key>
					<dict>
						<key>col</key>
						<integer>5</integer>
						<key>file</key>
						<integer>0</integer>
						<key>line</key>
						<integer>19</integer>
					</dict>
					<key>message</key>
					<string>array index 10 is past the end of the array (that has type 'int[5]')</string>
				</dict>
			</array>
			<key>type</key>
			<string>clang-tidy</string>
		</dict>
		<dict>
			<key>category</key>
			<string>clang</string>
			<key>check_name</key>
			<string>clang-diagnostic-error</string>
			<key>description</key>
			<string>expected ';' at end of declaration</string>
			<key>issue_hash_content_of_line_in_context</key>
			<string>d1e2722d18925ba9b4f6c1f9bbec1a22</string>
			<key>location</key>
			<dict>
				<key>col</key>
				<integer>19</integer>
				<key>file</key>
				<integer>0</integer>
				<key>line</key>
				<integer>30</integer>
			</dict>
			<key>notes</key>
			<array>
				<dict>
					<key>location</key>
					<dict>
						<key>col</key>
						<integer>19</integer>
						<key>file</key>
						<integer>0</integer>
						<key>line</key>
						<integer>30</integer>
					</dict>
					<key>message</key>
					<string>; (fixit)</string>
				</dict>
			</array>
			<key>path</key>
			<array>
				<dict>
					<key>depth</key>
					<integer>0</integer>
					<key>kind</key>
					<string>event</string>
					<key>location</key>
					<dict>
						<key>col</key>
						<integer>19</integer>
						<key>file</key>
						<integer>0</integer>
						<key>line</key>
						<integer>30</integer>
					</dict>
					<key>message</key>
					<string>expected ';' at end of declaration</string>
				</dict>
			</array>
			<key>type</key>
			<string>clang-tidy</string>
		</dict>
	</array>
	<key>files</key>
	<array>
		<string>files/test7.cpp</string>
	</array>
	<key>metadata</key>
	<dict>
		<key>analyzer</key>
		<dict>
			<key>name</key>
			<string>clang-tidy</string>
		</dict>
		<key>generated_by</key>
		<dict>
			<key>name</key>
			<string>report-converter</string>
			<key>version</key>
			<string>x.y.z</string>
		</dict>
	</dict>
</dict>
</plist>
