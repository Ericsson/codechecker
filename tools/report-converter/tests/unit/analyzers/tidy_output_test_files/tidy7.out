files/test7.cpp:14:20: warning: use nullptr [modernize-use-nullptr]
   14 |     int* nullPtr = NULL;
      |                    ^~~~
      |                    nullptr
files/test7.cpp:19:5: warning: array index 10 is past the end of the array (that has type 'int[5]') [clang-diagnostic-array-bounds]
   19 |     arr[10] = 42;
      |     ^   ~~
files/test7.cpp:17:5: note: array 'arr' declared here
   17 |     int arr[5];
      |     ^
files/test7.cpp:30:19: error: expected ';' at end of declaration [clang-diagnostic-error]
   30 |     int errVar = 3
      |                   ^
      |                   ;
